pkg/proxy/userspace/port_allocator.go

```diff
@@ -118,18 +118,14 @@ func (r *rangeAllocator) nextFreePort() int {
 	}
 
 	// search sequentially
-	for i := j + 1; i < r.Size; i++ {
-		if b := r.used.Bit(i); b == 0 {
-			r.used.SetBit(&r.used, i, 1)
-			return i + r.Base
-		}
-	}
-	for i := 0; i < j; i++ {
-		if b := r.used.Bit(i); b == 0 {
-			r.used.SetBit(&r.used, i, 1)
-			return i + r.Base
+	for i := 0; i < r.Size; i++ {
```

Comment:
if we're really determined to combine the loops, then we should probably get rid of the `if` above as well:
```go
	// search sequentially
	for i := j; i < r.Size + j; i++ {
		next := i % r.Size
		if r.used.Bit(next) == 0 {
			r.used.SetBit(&r.used, next, 1)
			return next + r.Base
		}
	}
```